crates/kubelet/src/volume/projected.rs

```diff
@@ -0,0 +1,250 @@
+use std::path::Path;
+
+use either::Either;
+use k8s_openapi::api::authentication::v1::{BoundObjectReference, TokenRequest, TokenRequestSpec};
+use k8s_openapi::api::core::v1::{
+    ConfigMapVolumeSource, DownwardAPIVolumeSource, Pod as KubePod, SecretVolumeSource,
+    Volume as KubeVolume, VolumeProjection,
+};
+use k8s_openapi::Resource;
+use tracing::warn;
+
+use super::*;
+
+/// A type that can manage a Secret volume with mounting and unmounting support
+pub struct ProjectedVolume {
+    vol_name: String,
+    volumes: Vec<super::VolumeRef>,
+    service_accounts: Vec<ServiceAccountSource>,
+    mounted_path: Option<PathBuf>,
+}
+
+struct ServiceAccountSource {
+    file_name: String,
+    service_account_name: String,
+    namespace: String,
+    client: kube::Client,
+    audience: String,
+    expiration_time: i64,
+    pod_name: String,
+    pod_uid: String,
+}
+
+impl ServiceAccountSource {
+    async fn mount_at(&mut self, path: impl AsRef<Path>) -> anyhow::Result<()> {
+        // As far as I can tell, this is the only way to access the token subresource on service accounts
+        let (req, _) = TokenRequest::create_namespaced_service_account_token(
```

Comment:
Should the second return parameter be handled? It appears to return type StatusCode... I'm wondering if we need to handle that as well as the Result (which we implicitly handle on line 53).

https://docs.rs/k8s-openapi/0.12.0/k8s_openapi/api/authentication/v1/struct.TokenRequest.html

Reply:
Nope, it took me a sec to grok this as well. It returns a function that can create an error type from a status code. All I need here is the raw request I can pass into the `kube` library, which does all of its own error conversion stuff for me